src/hooks/useAnimatedRender.js

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react'
+import { getClosestFocusableParent } from '../utilities/focus'
+
+/**
+ *
+ * @param {boolean} show Whether the target component should be shown or not
+ * @param {Object} toAnimateRef The ref to the node to animate out
+ * @param {Object} toFocusRef If provided, the ref to the node to focus
+ *
+ * @returns {Array} whether to render and onAnimationEnd callback
+ */
+export default function useAnimatedRender(show, toAnimateRef, toFocusRef) {
+  const [shouldRender, setRender] = useState(show)
+
+  useEffect(() => {
+    if (show) {
+      setRender(true)
+    }
+  }, [show])
+
+  function onAnimationEnd(e) {
+    if (e.target === toAnimateRef.current) {
+      if (!show) {
+        setRender(false)
+
+        if (toAnimateRef.current) {
```

Comment:
I think it would never be false? if `e.target === toAnimateRef.current` is true, and I think `target` would never be undefined/null (this is the place I have not checked, but my intuition tells me that :smile:), then there should be no need to have this additional check

Reply:
Good catch!